.github/workflows/system_tests.yml

```diff
@@ -0,0 +1,68 @@
+name: System Tests
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      RACK_ENV: test
+      RAILS_ENV: test
+      NODE_ENV: test
+      PORT: 3000
+      DISABLE_SPRING: 1
+      CAPYBARA_APP_HOST: "" # Explicitly set a blank value to override our default Docker container setup
+      NEW_RELIC_ENABLED: false
+      PAYLOAD_SECRET: amazing-payload-secret
+      AUTH_SECRET: amazing-authjs.secret # Add by executing `npx auth`. Read more: https://cli.authjs.dev
+      OIDC_ISSUER: http://wca_on_rails:3000 # Usually your local (Docker-internal!) Rails root URL
+      OIDC_CLIENT_ID: create-your-own-oauth # Create an OAuth application in the monolith with (at least!) `public`, `oidc`, `profile` and `email` scopes
+      OIDC_CLIENT_SECRET: create-your-own-oauth # See above
+      DATABASE_URI: file:./next-frontend.db
+    steps:
+      - uses: actions/checkout@v4
+      - uses: fregante/setup-git-user@v2 # set up dummy user.name and user.email in git so that Overcommit doesn't explode
+      - name: Set up ImageMagick and Libvips # This is not included in GitHub runners by default anymore
+        run: |
+          sudo apt update
+          sudo apt install --no-install-recommends imagemagick libvips
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+      - name: Install fresh Bundler packages
+        run: bin/bundle install
+      - name: Set up Corepack/Yarn
+        run: corepack enable # this allows NPM to use its own Yarn. It is crucial that this is run BEFORE the Node setup!
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'yarn'
+      - name: Install fresh Yarn packages
+        run: bin/yarn install
+      - name: Run Overcommit commit hooks
+        run: |
+          gem install overcommit
+          overcommit --sign
+          overcommit --sign pre-commit
+          overcommit --run
```

Comment:
I don't think our backend linter should run for frontend integration tests. We have a separate GitHub Action that cares about backend tests.